builtin/providers/kubernetes/structures_pod.go

```diff
@@ -0,0 +1,112 @@
+package kubernetes
+
+import (
+	"k8s.io/kubernetes/pkg/api/v1"
+)
+
+// Flatteners
+func flattenPodSpec(in v1.PodSpec) []interface{} {
+	att := make(map[string]interface{})
+	if in.ActiveDeadlineSeconds != nil {
+		att["active_deadline_seconds"] = *in.ActiveDeadlineSeconds
+	}
+	if in.DNSPolicy != "" {
+		att["dns_policy"] = in.DNSPolicy
```

Comment:
Looks like the terraform core is somehow taking care of serializing the right type ? Seems like magically working. I will check this again and put the cast.